backend/dataall/db/api/share_object.py

```diff
@@ -258,13 +258,23 @@ def approve_share_object(
 
         share.status = ShareObjectStatus.Approved.value
 
-        ResourcePolicy.attach_resource_policy(
-            session=session,
-            group=share.principalId,
-            permissions=permissions.DATASET_READ,
-            resource_uri=dataset.datasetUri,
-            resource_type=models.Dataset.__name__,
-        )
+        # GET TABLES SHARED AND APPROVE SHARE FOR EACH TABLE
```

Comment:
I will need to make updates to this piece with the new design of the sharing state machine, but with this implementation I think it is fine